### PR TITLE
Revert "Revert "Support experimental document formatting in-editor with Rubyfmt (#3551)" (#3553)"

### PR DESCRIPTION
**Pull request by penelope-stripe**
Fixes the master build for the mac linking we think
### Test plan
This failed locally then started passing locally after this change

### CODE REVIEW

**Comment by jez** [comment]
> Test plan

@penelope-stripe Can you change `.buildkite/pipeline.yaml` to run the build on macOS for this branch?

**Comment by jez** [comment]
https://github.com/sorbet/sorbet/blob/02dd827b4854b5cf24fdaef4f58118fd6955c3c0/.buildkite/pipeline.yaml#L29

^ delete this line, push to this PR, and then i can unblock the build and check that it passes for master

**Comment by jez** [comment]
you can do it in a PR based off of this PR if you don't want to have to worry about making that change, waiting for a build, and then reverting that change.

File: .buildkite/pipeline.yaml
Comment (jez) [inline .buildkite/pipeline.yaml:29]: Don't forget to remove this before merging.

Comment (jvilk) [reply .buildkite/pipeline.yaml:29]: I'm planning to get a re-review from trevor before merging. Am removing this and commenting one of the bazel changes.

**Comment by jvilk** [comment]
@penelope-stripe heads up -- I'm going to do a bit of rebasing/squashing so this is easier for trevor (and me, tbh) to review.

**Comment by jvilk** [comment]
@elliottt this is ready for review. You can review commit by commit. There are two changes -- one is a bugfix + test the other is the fix to the CI issue.

**Review by elliottt — APPROVED** [review]
Looking forward to using this!

**Pull request merged**
